src/faebryk/library/NetTie.py

```diff
@@ -0,0 +1,51 @@
+# This file is part of the faebryk project
+# SPDX-License-Identifier: MIT
+
+import logging
+from enum import Enum
+
+import faebryk.library._F as F
+from faebryk.core.module import Module
+from faebryk.core.moduleinterface import ModuleInterface
+from faebryk.library.Electrical import Electrical
+from faebryk.library.KicadFootprint import KicadFootprint
+from faebryk.libs.picker.picker import has_part_picked_remove
+
+logger = logging.getLogger(__name__)
+
+
+class NetTie[T: ModuleInterface](Module):
+    class Size(float, Enum):
+        _2_0MM = 2.0
+        _0_5MM = 0.5
+
+    unnamed: list[T]
+
+    def __init__(
+        self,
+        width: Size,
+        interface_type: type[T] = Electrical,
+    ) -> None:
+        super().__init__()
+
```

Comment:
__init__ in fab ll is purely for argument passing
split into __init__ and __preinit__

```python
    def __init__(
        self,
        width: Size,
        interface_type: type[T] = Electrical,
    ) -> None:
        super().__init__()
        self._interface_type = interface_type
        self._width = width

   def __preinit__(self): ...
```

Reply:
Wait though, if `__preinit__` is called before `__init__` then how can I access `_width` during my static configuration?

Reply:
The order is __init__, __preinit__, __postinit__
I can see where the confusion comes from.
The pre/post refers to pre/post base class construction